app/policies/student_flashcard_policy.rb

```diff
@@ -0,0 +1,28 @@
+class StudentFlashcardPolicy < ApplicationPolicy
+  class Scope < Scope
+    def resolve
+      scope.all
+    end
+  end
+
+    def index?
+      true
+    end
+
+    def show?
+      true
+    end
+
+    def create?
+      true
+    end
+
+    def update?
+      record.teacher == user.teacher
```

Comment:
record is student_flashcard, which doesn't have a teacher
record.flashcard_template.teacher == user.teacher